tests/system_tests/epics/adsim/test_adsim_system.py

```diff
@@ -35,17 +36,23 @@
 TIMEOUT = 10.0
 
 
-@pytest.fixture(scope="module", autouse=True)
+@pytest.fixture(scope="module")
 def with_env():
-    with patch.dict(
-        os.environ,
-        {
-            "EPICS_CA_NAME_SERVERS": "127.0.0.1:9064",
-            "EPICS_PVA_NAME_SERVERS": "127.0.0.1:9075",
-        },
-        clear=True,
-    ):
+    target_env = {
+        "EPICS_CA_NAME_SERVERS": "127.0.0.1:9064",
+        "EPICS_PVA_NAME_SERVERS": "127.0.0.1:9075",
+    }
+    # Save original values to restore later
+    original_env = {key: os.environ.get(key, "") for key in target_env}
+
+    # Set the test values
+    os.environ.update(target_env)
+
+    try:
         yield
+    finally:
+        # Restore original values (or delete if originally unset)
+        os.environ.update(original_env)
```

Comment:
Now we've worked out these need to be run in a subprocess, I think you can go back to using `patch.dict`